scripts/check.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python3
+import argparse
+import os
+import subprocess
+import sys
+import json
+
+PROD_SIGNING_KEY_FPR="pubkeys/prod.key"
+DEV_SIGNING_KEY_FPR="pubkeys/test.key"
+RPM_DIR="workstation"
+
+def verify_sig_rpm(path, dev=False):
+
+    key_path = ""
+    if dev:
+        key_path = DEV_SIGNING_KEY_FPR
+    else:
+        key_path = PROD_SIGNING_KEY_FPR
+    try:
+        subprocess.check_call(["rpmkeys", "--import", key_path])
+    except subprocess.CalledProcessError as e:
+        fail("Error importing key: {}".format(str(e)))
+
+    # Check the signature
+    try:
+        output = subprocess.check_output(["rpm", "--checksig", path])
+        # rpm --checksig returns 0 if there is *no* signature. I couldn't
+        # find a way other than parsing stdout
+        print(str(output))
+        if "digests signatures OK" not in str(output):
+            fail("Signture verification failed for {}:{}".format(path, output))
+    except subprocess.CalledProcessError as e:
+        fail("Error checking signature: {}".format(str(e)))
+
+
+def verify_all_rpms(dev=False):
+    for root, dirs, files in os.walk(RPM_DIR):
+        for name in files:
+            verify_sig_rpm(os.path.join(root,name), dev)
+
+def remove_keys_in_rpm_keyring():
+    try:
+        subprocess.check_call(["rpm", "--erase", "--allmatches", "gpg-pubkey"])
+    except subprocess.CalledProcessError:
+        pass
+
+def fail(msg):
+    print(msg, file=sys.stderr)
+    sys.exit(1)
+
+def main():
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument('--dev', action='store_true', default=False)
+    parser.add_argument('--verify', action='store_true', default=True)
+    parser.add_argument('--all', action='store_true', default=False)
+    parser.add_argument('packages', type=str, nargs='*', help='Files to sign/verify')
```

Comment:
Out of scope for a CI-only script, but in order to guard against user error, we should handle the no-args invocation of the script:

```
$ ./scripts/check.py 
Namespace(all=False, dev=False, packages=[], verify=True)
rpm: RPM should not be used directly install RPM packages, use Alien instead!
rpm: However assuming you know what you are doing...
error: package gpg-pubkey is not installed
$ echo $?
0
```

We definitely want that to exit non-zero, no? Can't do `nargs='*'` -> `nargs='+'`, since `--all` means `packages` is optional, but an `if not args.all and len(packages) < 1: fail()` would help.